### PR TITLE
Make SiStripDCSFilter a stream module

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

In the context of #25090, this PR makes SiStripDCSFilter a stream EDFilter instead of legacy, and in addition
* uses ESGetToken
* removes unnecessary members of SiSTripDCSStatus class

#### PR validation:

Limited matrix runs.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

File: CalibTracker/SiStripCommon/src/SiStripDCSStatus.cc [hunk 1]
```diff
@@ -97,13 +95,10 @@ bool SiStripDCSStatus::getStatus(edm::Event const& e, edm::EventSetup const& eSe
 //
 void SiStripDCSStatus::initialise(edm::Event const& e, edm::EventSetup const& eSetup) {
   //Retrieve tracker topology from geometry
-  edm::ESHandle<TrackerTopology> tTopoHandle;
-  eSetup.get<TrackerTopologyRcd>().get(tTopoHandle);
-  const TrackerTopology* const tTopo = tTopoHandle.product();
+  const auto& tTopo = eSetup.getData(tTopoToken_);
+  const auto& fedCabling_ = eSetup.getData(fedCablingToken_);
```
Comment (makortel) [inline CalibTracker/SiStripCommon/src/SiStripDCSStatus.cc]: Pardon my ignorance, but given that `initialise()` is run once per job (and per EDM stream, or at least until an event with a valid `FEDRawDataCollection` has been encountered), is `SiStripFedCablingRcd` guaranteed to have a long-enough IOV that it won't get an update within a single job?

Comment (mmusich) [reply CalibTracker/SiStripCommon/src/SiStripDCSStatus.cc]: The FED cabling is changed maybe once / twice per year (if necessary).

**Comment by pohsun** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**